lib/client/util/__mocks__/Geolocation.js

```diff
@@ -0,0 +1,37 @@
+var path = require.requireActual('../Geolocation');
+
+var geoMock = jest.genMockFromModule('../Geolocation');
+
+var _coords = null;
+var _enabled = false;
+function __setLocation(lat, lon) {
+  _enabled = true;
+  _coords = { latitude: lat, longitude: lon };
+}
+
+function __enable() {
+  _enabled = true;
+}
+
+function __disable() {
+  _coords = null;
+  _enabled = false;
+}
+
+function getCurrentPosition(successCallback, errorCallback) {
+  if(_enabled) {
+    if(_coords) {
+      successCallback({ coords: _coords });
+    }
```

Comment:
Yes, even though the API that you use operates synchronously you can still test code that consumes it synchronously.  The code that provides a callback is not *intrinsically* asynchronous.  However, I do understand your assertion that if a piece of code is async, its mock should be too.

You don't have to use the function calls in Jest to get the timers to run, but how would you force an operation to occur in time before your assertion is checked?  With a timer, your callback is placed on the global queue, and your test will finish executing (and the assertion fail) before that code is executed unless you can force queue to be flushed ahead of time.  (It's very possible I'm missing something here, and this is the crux of my issue.)

For me, it comes down to a matter of complexity and cognitive load.  Given a mock with async behavior that I didn't create, I need to know which calls are async and be sure to make the appropriate flush call in each test.  This makes my tests more complex and I have to know about the internals of the mock (rather than just the API).  Worse, as I start building functional tests, the async nature of the mock starts to infect tests in other modules as well.

I think this is worth more discussion, maybe over on the Slack channel (http://tech256.com/)?  I'd rather spend the time now coming to establish good standards before moving forward, but I defer that decision to you.